### PR TITLE
Delete

### PR DESCRIPTION
**Pull request by QuanXuuu**

### CODE REVIEW

File: src/pages/RecipeDetail.js [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
   const params = useParams();
+  const navigate = useNavigate();
   const recipe = recipes.find(recipe => recipe.id === params.id);
   const ingredientItems = recipe.ingredients.map((ingredient, index) => <li key={index}>{ingredient}</li>);
 
+  const [isDialogueShown, setIsDialogueShown] = useState(false);
+
+  const showDialogue = () => {
+    setIsDialogueShown(true);
+    console.log(isDialogueShown);
```
Comment (macfrei) [inline src/pages/RecipeDetail.js:21]:
```suggestion
```

File: src/pages/RecipeDetail.js [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+  };
+
+  const hideDialogue = () => {
+    setIsDialogueShown(false);
+    console.log(isDialogueShown);
```
Comment (macfrei) [inline src/pages/RecipeDetail.js:26]:
```suggestion
```

File: src/pages/RecipeDetail.js [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
+  };
```
Comment (macfrei) [inline src/pages/RecipeDetail.js:27]: You could actually combine these two in one function though:
```suggestion
  const toggleDialogue = () => {
    setIsDialogueShown(!isDialogueShown);
  };
```

**Review by macfrei — COMMENTED** [review]
Good job, added a few comments :)

File: src/pages/RecipeDetail.js [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
 import {FiAlignJustify} from 'react-icons/fi';
-import {useParams} from 'react-router-dom';
+import {useParams, useNavigate} from 'react-router-dom';
 import styled from 'styled-components';
 
-export default function RecipeDetail({recipes}) {
+import Dialogue from '../components/Dialogue/Dialogue';
+
+export default function RecipeDetail({recipes, onDeleteRecipe}) {
   const params = useParams();
```
Comment (ahohnsen) [inline src/pages/RecipeDetail.js:12]: Here you can also destructure the id directly:

```suggestion
  const {id} = useParams();
```

File: src/pages/RecipeDetail.js [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
+  const navigate = useNavigate();
   const recipe = recipes.find(recipe => recipe.id === params.id);
```
Comment (ahohnsen) [inline src/pages/RecipeDetail.js:14]: Then you could change this to:

```suggestion
  const recipe = recipes.find(recipe => recipe.id === id);
```

File: src/pages/RecipeDetail.js [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
+  const [isDialogueShown, setIsDialogueShown] = useState(false);
+
+  const toggleDialogue = () => {
+    setIsDialogueShown(!isDialogueShown);
+  };
+
+  const handleDelete = () => {
+    onDeleteRecipe(params.id);
+    navigate('/');
+  };
```
Comment (ahohnsen) [inline src/pages/RecipeDetail.js:26]: See above :) 

```suggestion
  const handleDelete = () => {
    onDeleteRecipe(id);
    navigate('/');
  };
```

**Review by ahohnsen — APPROVED** [review]
Hey Quan, 

Great job! 💪🏼 🚀 I only have a suggestion to make your code a tiny bit shorter 😄

**Pull request merged**